### PR TITLE
Realtime transport auth fixed (RTC1)

### PR DESCRIPTION
**Pull request by ricardopereira**
- `RTC1` is passing.

Next on new PR:
 - `ARTError` need to be a `NSError`.
 - Channels are not working.

### CODE REVIEW

File: ably-ios/ARTWebSocketTransport.m [hunk 1]
```diff
@@ -127,7 +80,81 @@ - (void)send:(ARTProtocolMessage *)msg {
...
+            ARTWebSocketTransport *selfStrong = selfWeak;
+            if (!selfStrong) return;
+
+            if (error) {
+                [selfStrong.logger error:@"ARTWebSocketTransport: token auth failed with %@", error.description];
+                [selfStrong.delegate realtimeTransportFailed:selfStrong];
+                return;
+            }
+
+            NSURLQueryItem *accessTokenParam = [NSURLQueryItem queryItemWithName:@"access_token" value:decodeBase64(tokenDetails.token)];
```
Comment (ricardopereira) [inline ably-ios/ARTWebSocketTransport.m]: @paddybyers Correct? I'm decoding the access token (`decodeBase64`).

File: ably-ios/ARTWebSocketTransport.m [hunk 2]
```diff
@@ -127,7 +80,81 @@ - (void)send:(ARTProtocolMessage *)msg {
...
+            [self.logger error:@"ARTWebSocketTransport: recovery string is malformed, ignoring: '%@'", options.recover];
+        }
+    }
+    else if (options.resumeKey != nil) {
+        NSURLQueryItem *resumeKeyParam = [NSURLQueryItem queryItemWithName:@"resume" value:options.resumeKey];
+        queryItems = [queryItems arrayByAddingObject:resumeKeyParam];
+
+        NSURLQueryItem *connectionSerialParam = [NSURLQueryItem queryItemWithName:@"connection_serial" value:[NSString stringWithFormat:@"%lld",options.connectionSerial]];
+        queryItems = [queryItems arrayByAddingObject:connectionSerialParam];
+    }
```
Comment (ricardopereira) [inline ably-ios/ARTWebSocketTransport.m]: @paddybyers Please review.

**Comment by ricardopereira** [comment]
Merge when it's possible. Thank you,

File: ably-ios/ARTWebSocketTransport.m (hunk 1 above)
Comment (paddybyers) [inline ably-ios/ARTWebSocketTransport.m]: There shouldn't be any need to decode the token

**Comment by paddybyers** [comment]
Can you check that comment above about the token? You should treat the token string as opaque.

**Comment by ricardopereira** [comment]
But the received token is encoded and then the it's not possible to establish connection to the realtime server.

**Comment by SimonWoolf** [comment]
@ricardopereira realtime provides it as a base64-encoded string, and expects back the same base64-encoded string. If it's not working unless you have a decode in there, I'd hazard a guess that there's code elsewhere that's mistakenly encoding it before sending it.

(The client lib can't do anything with it decoded anyway -- it won't be valid ascii when decoded, it's a binary)

**Comment by ricardopereira** [comment]
@SimonWoolf Ok, I will investigate. Thanks.

**Comment by ricardopereira** [comment]
@paddybyers @SimonWoolf Fixed. Encoder was misplaced on ARTAuth.

**Comment by paddybyers** [comment]
Thanks

**Pull request merged**